relayer-engine/src/walletMonitor/walletMonitorHarness.ts

```diff
@@ -0,0 +1,86 @@
+import { CHAIN_ID_SOLANA, isEVMChain, isTerraChain } from "@certusone/wormhole-sdk";
+import { ChainConfigInfo } from "../../packages/relayer-plugin-interface";
+import { getCommonEnv, CommonEnv, ExecutorEnv, getExecutorEnv, PrivateKeys, TokenMonitoringInfo } from "../config";
+import { getScopedLogger, ScopedLogger } from "../helpers/logHelper";
+import { sleep } from "../utils/utils"
+import { WalletBalance, pullAllSolanaBalances, pullAllEVMBalances, pullAllTerraBalances } from './balances';
+import { updateWalletBalances } from "./metrics";
+
+const ONE_MINUTE: number = 60000;
+
+let _logger: ScopedLogger;
+
+const logger = () => {
+  if (!_logger) _logger = getScopedLogger(["WalletMonitor"]);
+  return _logger;
+}
+
+export type ChainMonitoringConfig = {
+  chainInfo: ChainConfigInfo,
+  privateKeys: string[],
+}
+
+async function pullBalances(monitorConfig: ChainMonitoringConfig[], tokensToMonitor: TokenMonitoringInfo[]): Promise<WalletBalance[]> {
+  const balancePromises: Promise<WalletBalance[]>[] = [];
+
+  for (const { chainInfo, privateKeys } of monitorConfig) {
+    switch (true) {
+      case chainInfo.chainId === CHAIN_ID_SOLANA:
+        balancePromises.push(pullAllSolanaBalances(chainInfo, privateKeys));
+        break;
+
+      case isEVMChain(chainInfo.chainId):
+        balancePromises.push(...pullAllEVMBalances(chainInfo, privateKeys, tokensToMonitor));
+        break;
+
+      case isTerraChain(chainInfo.chainId):
+        pullAllTerraBalances(chainInfo, privateKeys, tokensToMonitor);
```

Comment:
`pullAllTerraBalances` uses internally `updateWalletBalances` instead of returning the result and handling that uniformly with other wallet updates

Reply:
yup, just as Giorgio mentions...
This is definetly not the best practive and we were worried about this with Gio, but didn't want to start a big refactor because we didn't yet understand the rationale of leaving some of the balances to be pulled async.
Ideally I would like to refactor this so that nothing is done async since we have a potential issue if the pulling of the balances turns into a long running task (longer than the period configured for the wallet monitor)

Reply:
let's confirm with jeff or git blame the pull terra balances and ask the original author if this is needed, if not, let's enforce consistency